From f2e5bd60ea8c6339d4fdc27bfeb8afa4166a7151 Mon Sep 17 00:00:00 2001
From: Michael Goffioul <michael.goffioul@lincor.com>
Date: Wed, 28 Apr 2021 16:07:15 -0400
Subject: [PATCH 1/2] Make rules using bison/flex depend on BISON, FLEX and M4
 variables

In Android S, prebuilt tools are now copied into intermediate
directories and the make variables are populated using these copied.
This means files generated by these tools must depend on the
corresponding variables to make sure the tool is installed/copied before
tying to it.

This has been done through trials and errors and it's possible some
rules have been missed.
---
 src/compiler/Android.glsl.gen.mk | 4 ++++
 src/freedreno/Android.ir3.mk     | 3 +++
 src/mesa/program/Android.mk      | 2 ++
 3 files changed, 9 insertions(+)

diff --git a/src/compiler/Android.glsl.gen.mk b/src/compiler/Android.glsl.gen.mk
index ba7cb429d02..83eef2160fa 100644
--- a/src/compiler/Android.glsl.gen.mk
+++ b/src/compiler/Android.glsl.gen.mk
@@ -76,17 +76,21 @@ define local-yy-to-cpp-and-h
 	rm -f $(@:$1=$(YACC_HEADER_SUFFIX))
 endef
 
+$(intermediates)/glsl/glsl_lexer.cpp: $(M4) $(LEX)
 $(intermediates)/glsl/glsl_lexer.cpp: $(LOCAL_PATH)/glsl/glsl_lexer.ll
 	$(call local-l-or-ll-to-c-or-cpp)
 
+$(intermediates)/glsl/glsl_parser.cpp: $(BISON)
 $(intermediates)/glsl/glsl_parser.cpp: $(LOCAL_PATH)/glsl/glsl_parser.yy
 	$(call local-yy-to-cpp-and-h,.cpp)
 
 $(intermediates)/glsl/glsl_parser.h: $(intermediates)/glsl/glsl_parser.cpp
 
+$(intermediates)/glsl/glcpp/glcpp-lex.c: $(M4) $(LEX)
 $(intermediates)/glsl/glcpp/glcpp-lex.c: $(LOCAL_PATH)/glsl/glcpp/glcpp-lex.l
 	$(call local-l-or-ll-to-c-or-cpp)
 
+$(intermediates)/glsl/glcpp/glcpp-parse.c: $(BISON)
 $(intermediates)/glsl/glcpp/glcpp-parse.c: $(LOCAL_PATH)/glsl/glcpp/glcpp-parse.y
 	$(call glsl_local-y-to-c-and-h)
 
diff --git a/src/freedreno/Android.ir3.mk b/src/freedreno/Android.ir3.mk
index 0fbb8c50ca6..ee833b4a39b 100644
--- a/src/freedreno/Android.ir3.mk
+++ b/src/freedreno/Android.ir3.mk
@@ -75,6 +75,7 @@ ir3_nir_trig_deps := \
 ir3_parser_deps := \
 	$(MESA_TOP)/src/freedreno/ir3/ir3_parser.y
 
+$(intermediates)/ir3/ir3_lexer.c: $(M4) $(LEX)
 $(intermediates)/ir3/ir3_lexer.c: $(ir3_lexer_deps)
 	@mkdir -p $(dir $@)
 	@echo "Gen Header: $(PRIVATE_MODULE) <= $(notdir $(@))"
@@ -88,11 +89,13 @@ $(intermediates)/ir3/ir3_nir_trig.c: $(ir3_nir_trig_deps)
 	@mkdir -p $(dir $@)
 	$(hide) $(MESA_PYTHON3) $< -p $(MESA_TOP)/src/compiler/nir > $@
 
+$(intermediates)/ir3/ir3_parser.c: $(BISON)
 $(intermediates)/ir3/ir3_parser.c: $(ir3_parser_deps)
 	@mkdir -p $(dir $@)
 	@echo "Gen Header: $(PRIVATE_MODULE) <= $(notdir $(@))"
 	$(hide) $(BISON) $< --name-prefix=ir3_yy --output=$@
 
+$(intermediates)/ir3/ir3_parser.h: $(BISON)
 $(intermediates)/ir3/ir3_parser.h: $(ir3_parser_deps)
 	@mkdir -p $(dir $@)
 	@echo "Gen Header: $(PRIVATE_MODULE) <= $(notdir $(@))"
diff --git a/src/mesa/program/Android.mk b/src/mesa/program/Android.mk
index 6b4e1916747..a9856418aae 100644
--- a/src/mesa/program/Android.mk
+++ b/src/mesa/program/Android.mk
@@ -63,12 +63,14 @@ LOCAL_SRC_FILES := \
 LOCAL_GENERATED_SOURCES := \
 	$(addprefix $(intermediates)/program/,$(generated_sources_basenames))
 
+$(intermediates)/program/program_parse.tab.c: $(BISON)
 $(intermediates)/program/program_parse.tab.c: $(LOCAL_PATH)/program_parse.y
 	$(mesa_local-y-to-c-and-h)
 
 $(intermediates)/program/program_parse.tab.h: $(intermediates)/program/program_parse.tab.c
 	@
 
+$(intermediates)/program/lex.yy.c: $(M4) $(LEX)
 $(intermediates)/program/lex.yy.c: $(LOCAL_PATH)/program_lexer.l
 	$(local-l-to-c)
 
-- 
2.30.2

